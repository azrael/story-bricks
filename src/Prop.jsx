import React from 'react';
import pt from 'prop-types';
import s from './story.css';
import p from './prop.css';

const Prop = ({ name, type, optional, defaultValue, children }) => <>
    <div className={p.type}>
        <span className={p.name}>
            {name}
        </span>

        {type && ': '}

        <span className={s.mono}>
            {optional && '?'}
            {type}
            {defaultValue !== void 0 && <> = {defaultValue}</>}
        </span>
    </div>

    {children && <div className={p.description}>
        {children}
    </div>}
</>;

Prop.displayName = 'Prop';

Prop.propTypes = {
    name: pt.string.isRequired,
    type: pt.string.isRequired,
    optional: pt.bool,
    defaultValue: pt.any
};

Prop.defaultProps = {
    optional: false
};

export default Prop;
