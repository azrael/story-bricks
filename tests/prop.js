import React from 'react';
import renderer from 'react-test-renderer';
import Prop from '../src/Prop.jsx';

it('Prop with required props only (name and type) renders correctly', () => {
    const tree = renderer
        .create(<Prop name='foo' type='bar'>test</Prop>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Prop with defaultValue renders correctly', () => {
    const tree = renderer
        .create(<Prop name='foo' type='bar' defaultValue='baz'>test</Prop>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Prop with optional renders correctly', () => {
    const tree = renderer
        .create(<Prop name='foo' type='bar' optional>test</Prop>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
