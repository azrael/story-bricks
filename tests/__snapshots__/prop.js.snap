// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prop with defaultValue renders correctly 1`] = `
Array [
  <div
    className="type"
  >
    <span
      className="name"
    >
      foo
    </span>
    : 
    <span
      className="mono"
    >
      bar
       = 
      baz
    </span>
  </div>,
  <div
    className="description"
  >
    test
  </div>,
]
`;

exports[`Prop with optional renders correctly 1`] = `
Array [
  <div
    className="type"
  >
    <span
      className="name"
    >
      foo
    </span>
    : 
    <span
      className="mono"
    >
      ?
      bar
    </span>
  </div>,
  <div
    className="description"
  >
    test
  </div>,
]
`;

exports[`Prop with required props only (name and type) renders correctly 1`] = `
Array [
  <div
    className="type"
  >
    <span
      className="name"
    >
      foo
    </span>
    : 
    <span
      className="mono"
    >
      bar
    </span>
  </div>,
  <div
    className="description"
  >
    test
  </div>,
]
`;
